pkg/controller/sidecar/sync.go

```diff
@@ -104,6 +105,39 @@ func (c *Controller) decommissionNode(ctx context.Context, svc *corev1.Service)
 	return nil
 }
 
+func (c *Controller) setHostIdLabel(ctx context.Context, svc *corev1.Service) error {
+	if c.hostId == nil {
+		scyllaClient, err := controllerhelpers.NewScyllaClientForLocalhost()
+		if err != nil {
+			return err
+		}
+
+		hostid, err := scyllaClient.LocalHostid(ctx, localhost, false)
+		if err != nil {
+			return fmt.Errorf("can't get a local host_id: %w", err)
+		}
+
+		if len(hostid) == 0 {
+			return fmt.Errorf("received an invalid host_id")
+		}
+
+		c.hostId = &hostid
+	}
+
+	_, err := c.kubeClient.CoreV1().Services(c.namespace).Patch(
```

Comment:
would it be cleaner to do an UPDATE here? so any of the preconditions like `deletionTimestamp != nil` holds?

Reply:
this one is technically a status so it is kind of allowed to be updated even with deletion timestamp set but update still seems more fitting (and easier / type safe) here